### PR TITLE
Added unary functions

### PR DESCRIPTION
**Pull request by MaxMac99**
- ln (l)
- exp (e)
- x^2 (s)
- sqrt (q)

### CODE REVIEW

File: lib.cpp [hunk 1]
```diff
@@ -32,6 +37,22 @@ int evaluate(const std::string &s) {
     case '-': change_stack_state(minus,    stack); break;
     case '*': change_stack_state(multiply, stack); break;
     case '/': change_stack_state(divide,   stack); break;
+    case 'l':
+        stack.push('0');
+        change_stack_state(ln, stack);
+        break;
+    case 'e':
+        stack.push('0');
+        change_stack_state(exp,    stack);
```
Comment (tfc) [inline lib.cpp]: we do now obviously have functions that use two stack variables and functions that only use one.

can you please:

- rename `change_stack_state` to `apply_binary_function(function, stack)`
- add nice type aliases (`using` clause) for binary and unary functions
- add another `apply_unary_function` function and use that in those cases

**Comment by tfc** [comment]
Github says that there are merge conflicts that need to be resolved first. please resolve them.

**Comment by MaxMac99** [comment]
How can there be merge conflicts? There are no new commits.

**Comment by tfc** [comment]
> How can there be merge conflicts? There are no new commits.

Yesterday in the evening i pushed the removal of the redundant `using` clause in the lib.cpp file. it seems you branched off on the parent commit.

**Comment by tfc** [comment]
We have two issues here:

- new merge conflicts
- running `nix-build` shows that the test *hangs* (as the pipeline does right now)

**Pull request merged**